tests/baselines/reference/jsxFactoryAndJsxFragmentFactoryNull.types

```diff
@@ -3,31 +3,24 @@
 === jsxFactoryAndJsxFragmentFactoryNull.tsx ===
 declare var h: any;
 >h : any
->  : ^^^
 
 <></>;
-><></> : any
->      : ^^^
+><></> : error

```

Comment:
This has changed to an error type and yet there are no errors reported for this test now. It looks suspicious to me but I'm not entirely sure if this is incorrect

Reply:
Yeah, jake is right, error types in tests with `.error` baselines print as `any`, since it's assumed to be unimportant. An `error` type in a test WITHOUT a `.error` baseline, conversely, usually means there was a silent error of some kind that you need to look in to, which is why it's printed differently! Now, I imagine this is just because we skip call resolution here, but we should probably be returning `anyType` here, rather than `errorType`, since there's no errors anywhere in the test. Not that it's *that* important.

Reply:
@jakebailey @weswigham I fixed it. It looks like it was a bug in the first place because there was an improper use of `||` (`errorType || anyType` is still errorType)

Reply:
I don't think you actually need to conditionally return any on error; error is functionally any. The baselines are more or less cosmetic imo and I wish we didn't special case the baseline printout...

Reply:
Hmmmm.... fixing the rest of these means issuing new errors or running into https://github.com/microsoft/TypeScript/issues/18217 at getJsxNamespaceAt`. I think the correct way to improve the error types in the baselines is to issue a "cannot resolve jsx namespace" error (since the namespaces in the test don't resolve in the way the compiler expects a JSX namespace to be properly defined), but I don't want to fix that in this regression-fixing PR (it could make things confusing).

Reply:
The `errorTypes` left in the baselines are unrelated to this change, or the previous PR.